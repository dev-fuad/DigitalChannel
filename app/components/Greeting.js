/**
 * Greeting.js
 * Author: Fuad Mohd. Firoz
 * Created: December 06, 2020
 *
 * @flow
 */

import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {useSelector} from 'react-redux';

const Greeting: () => React$Node = () => {
  const name = useSelector((state) => state.name);

  return <Text style={styles.name}>{!!name && `Hello, ${name}`}</Text>;
};

const styles = StyleSheet.create({
  name: {
    fontSize: 16,
    color: '#2A2A2A',
    marginVertical: 8,
    marginHorizontal: 12,
    textAlign: 'right',
  },
});

export default Greeting;
