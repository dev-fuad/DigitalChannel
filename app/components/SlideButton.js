/**
 * SlideButton.js
 * Author: Fuad Mohd. Firoz
 * Created: December 06, 2020
 *
 * @flow
 */

import React from 'react';
import {StyleSheet, Dimensions, View} from 'react-native';
import SwipeButton from 'rn-swipe-button';

const {height, width} = Dimensions.get('screen');

const CONTAINER_HEIGHT = height * 0.06;
const CONTAINER_WIDTH = width * 0.9;
const BORDER_RADIUS = width * 0.015;

type SlideButtonProps = {
  style?: Object,
  title: string,
  onSwipeSuccess?: Function,
};

const SlideButton = ({
  style,
  title,
  onSwipeSuccess = () => {},
}: SlideButtonProps): React$Node => {
  return (
    <View style={styles.container}>
      <SwipeButton
        containerStyles={styles.container}
        thumbIconStyles={styles.thumb}
        railBackgroundColor="transparent"
        railBorderColor="#34424A"
        railStyles={styles.rail}
        titleColor="#145999"
        height={CONTAINER_HEIGHT}
        width={CONTAINER_WIDTH}
        thumbIconBackgroundColor="#6FB1F7"
        swipeSuccessThreshold={60}
        title={title}
        onSwipeStart={() => console.log('Swipe started!')}
        onSwipeFail={() => console.log('Incomplete swipe!')}
        onSwipeSuccess={onSwipeSuccess}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: CONTAINER_HEIGHT,
    width: CONTAINER_WIDTH,
    borderRadius: BORDER_RADIUS,
    margin: 0,
  },
  rail: {
    height: CONTAINER_HEIGHT,
    borderRadius: BORDER_RADIUS,
    borderWidth: 0,
    margin: 0,
    backgroundColor: '#6FB1F7BB',
  },
  thumb: {
    borderRadius: BORDER_RADIUS,
    borderWidth: 0,
    marginVertical: 0,
    marginRight: -4,
  },
});

export default SlideButton;
