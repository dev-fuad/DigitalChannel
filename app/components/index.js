export {default as Greeting} from './Greeting';
export {default as Button} from './Button/Button';
