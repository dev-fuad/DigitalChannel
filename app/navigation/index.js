/**
 * App.js
 * Author: Fuad Mohd. Firoz
 * Created: December 05, 2020
 *
 * @flow
 */

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';
import {PopUp} from '../modules/PopUp';
import {ButtonsScreen} from '../modules/buttons';
import {HomeScreen} from '../modules/home';
import {WelcomeScreen} from '../modules/welcome';

export const Screens = {
  homeStack: 'HomeStack',
  home: 'Home',
  welcome: 'Welcome',
  buttons: 'Buttons',
  popUp: 'PopUp',
};

const HomeStack = createStackNavigator();
const AppStack = createStackNavigator();

const HomeNavigation: () => React$Node = () => {
  return (
    <HomeStack.Navigator>
      <HomeStack.Screen name={Screens.home} component={HomeScreen} />
      <HomeStack.Screen name={Screens.welcome} component={WelcomeScreen} />
      <HomeStack.Screen name={Screens.buttons} component={ButtonsScreen} />
    </HomeStack.Navigator>
  );
};

const AppNavigation: () => React$Node = () => {
  return (
    <NavigationContainer>
      <AppStack.Navigator
        mode="modal"
        screenOptions={{
          headerShown: false,
          cardStyle: {backgroundColor: 'transparent'},
          cardOverlayEnabled: true,
        }}>
        <AppStack.Screen name={Screens.homeStack} component={HomeNavigation} />
        <AppStack.Screen name={Screens.popUp} component={PopUp} />
      </AppStack.Navigator>
    </NavigationContainer>
  );
};

export default AppNavigation;
