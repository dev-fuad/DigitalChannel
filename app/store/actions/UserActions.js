/**
 * UserActions.js
 * Author: Fuad Mohd. Firoz
 * Created: December 06, 2020
 *
 * @flow
 */

import type {UserAction} from '../../types/store';

export const updateUser = (name: string): UserAction => {
  return {type: 'UPDATE_NAME', payload: name};
};
