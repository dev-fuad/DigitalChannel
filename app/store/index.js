/**
 * index.js
 * Author: Fuad Mohd. Firoz
 * Created: December 06, 2020
 *
 * @flow
 */

import {createStore} from 'redux';
import type {Store} from '../types/store';
import reducer from './reducers/UserReducer';

const store: Store = createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__?.(),
);

export default store;
