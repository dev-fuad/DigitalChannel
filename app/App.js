/**
 * App.js
 * Author: Fuad Mohd. Firoz
 * Created: December 05, 2020
 *
 * @flow
 */

import React from 'react';
import {Provider} from 'react-redux';
import AppNavigation from './navigation';
import store from './store';

const App: () => React$Node = () => {
  return (
    <Provider store={store}>
      <AppNavigation />
    </Provider>
  );
};

export default App;
