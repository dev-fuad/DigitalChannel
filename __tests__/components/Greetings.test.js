import {act, cleanup, render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {Greeting} from '../../app/components';
import {updateName} from '../../app/store/actions/UserActions';
import reducer from '../../app/store/reducers/UserReducer';

const store = createStore(reducer);
store.dispatch(updateName('Chris'));

describe('Greeting', () => {
  afterEach(cleanup);

  it('should show current name', () => {
    const {getByText} = render(
      <Provider store={store}>
        <Greeting />
      </Provider>,
    );
    const chris = getByText(/Chris/);
    expect(chris).toBeDefined();
  });

  it('should show updated name', () => {
    act(async () => {
      store.dispatch(updateName('Dave'));
      const {findByText} = render(
        <Provider store={store}>
          <Greeting />
        </Provider>,
      );
      const chris = await findByText(/Dave/);
      expect(chris).toBeDefined();
    });
  });

  it('should match snapshot', () => {
    const {toJSON} = render(
      <Provider store={store}>
        <Greeting />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
