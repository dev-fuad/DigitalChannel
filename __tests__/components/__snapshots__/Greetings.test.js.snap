// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#2A2A2A",
      "fontSize": 16,
      "marginHorizontal": 12,
      "marginVertical": 8,
      "textAlign": "right",
    }
  }
>
  Hello, Dave
</Text>
`;
